### DISCUSSION TITLE
Метод error у EntityState. Exception или Object или ..?

### DISCUSSION THREAD

**Discussion by IlyaMax** [surfstudio/SurfGear#209] [Ideas]
```
/// State of some logical entity
class EntityState<T> {
  /// Data of entity
  final T? data;

  /// State is loading
  final bool isLoading;

  /// State has error
  final bool hasError;

  /// Error from state
  final Object? error;
  ....
}
```
Поле Object должно в теории использоваться для того чтобы передавать Error в стейт. Может сделать ErrorWrapper какой-нибудь у которого будет два конструктора fromError и fromException? Тогда в еррор нельзя будет положить что-то кроме ошибки.

**Comment by HazarD2311** [comment]
Там в более ранних версиях он и был. Даже название как раз ErrorWrapper :)
Но мы от него отказались из частых неверных кастов к нему, учитываю что мы теперь еще и Dio напрямую используем.

Можно в принципе попробовать вернуть и аккуратно ошибки мапить, но как бы это не надоело

**Reply by r3tam** [reply]
Поле error - это поле для Exception'ов. По-хорошему Error не должны ловится во время выполнения. По сути отлавливаем мы только Exception'ы.
